### PR TITLE
add callback to g_BPInitialBudget and g_BPBudgetPerMiner

### LINKED ISSUE #1124: Available buildpoints do not change with cvar change
Changing g_BPBudgetPerMiner or g_BPInitialBudget only take effect after a leech/drill is build.

### PR DESCRIPTION
**Pull request by Gireen**
Fix #1124

### CODE REVIEW

**Comment by slipher** [comment]
Oh, if it's the case that the calculation works fine but just needs to be updated, you could use a cvar callback. You can make a function get called when the value changes. E.g. 
```
Cvar::Callback<Cvar::Cvar<int>> g_lol("g_lol", "description", 0, 50, [](int lol) { Log::Notice("lol = %d", lol); });
```

Looks like there are 3 obscure features that depend on cvars being in `gameCvarTable`, but none of them are used by the two BP cvars so it's fine to convert them to new-style (C++ API) cvars.

**Comment by Gireen** [comment]
the callback required more changes but that works really better.

File: src/sgame/sg_main.cpp [hunk 1]
```diff
@@ -107,8 +107,22 @@ vmCvar_t           pmove_accurate;
...
 vmCvar_t           g_maxNameChanges;
 
-vmCvar_t           g_buildPointInitialBudget;
-vmCvar_t           g_buildPointBudgetPerMiner;
+Cvar::Callback<Cvar::Cvar<int>> g_buildPointInitialBudget(
+		"g_BPInitialBudget",
+		"Initial Budget",
+		Cvar::SERVERINFO,
+		DEFAULT_BP_INITIAL_BUDGET,
+		[](int budget) {
```
Comment (slipher) [inline src/sgame/sg_main.cpp]: Remove the name of the unused variable (so like `[](int) {`... )

**Comment by slipher** [comment]
Please squash the commits into one, remove the unused variable names and LGTM otherwise.

**Comment by slipher** [comment]
You can merge it, right?

**Pull request merged**